### DISCUSSION TITLE
Adding a file to another model as a submodule

### DISCUSSION THREAD

**Discussion by edsousa84** [leozide/leocad#779] [Q&A] [answered]
Is it possible to add another file to leocad as a submodule instead of creating a new one from scratch?

**Comment by edsousa84** [comment]
I found a solution I copy pasted the content of the file to newly created submodules code blocks of the main model.

**Reply by nathaneltitane** [reply]
You can create submodels and assemble them as parts in the main model.

**Reply by edsousa84** [reply]
That was exacly what I wanhted to do my problem was HOw to add them in the first place. From a alredy created ldr file.
But As I stated I manage to make it work by:
1) Creating the new submodules (empy ones
2) Edit the ldr file (on a text editor) and copy the model I wahnt to add to the correct zones of the sub modules created.
3) Save the file and reopen on Leocad

Some time I nedded to fix one or oder line of the ldr file.

**Comment by leozide** [comment, marked as answer]
It works but there's no UI to do this, you have to edit the file by hand and replace the part name with the file name then it will load correctly

**Reply by nathaneltitane** [reply]
yes there is. look into the models menu... submodels entry

**Reply by leozide** [reply]
that's not adding another file, it's adding another model

**Reply by nathaneltitane** [reply]
but if you treat a model as just another part, it works, doesn't it?

leocad does not distinguish in between the dat ldr and mpd file extensions...

**Reply by edsousa84** [reply]
> but if you treat a model as just another part, it works, doesn't it?
> 
> leocad does not distinguish in between the dat ldr and mpd file extensions...

I did not find a way to make it work like that.
I manage to create new sub modules and as lezide stated edited the files by adding the models I had into it.

> It works but there's no UI to do this, you have to edit the file by hand and replace the part name with the file name then it will load correctly

That was exacly what I did

**Reply by edsousa84** [reply]
> but if you treat a model as just another part, it works, doesn't it?
> 
> leocad does not distinguish in between the dat ldr and mpd file extensions...

Sorry I did not understand what you meant
